### PR TITLE
update support for http redirection for workflows 

### PR DESCRIPTION
**Pull request by chamilaadhi**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-  update support for http redirection in API publisher. change the status code to 202 for pending tasks
-  update support for http redirection in API store. change the status code to 202 for pending tasks

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] ** Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] ** Are all UI and API inputs run through forms or serializers?**
- [ ] ** Are all external inputs validated and sanitized appropriately?**
- [ ] ** Does all branching logic have a default case?**
- [ ] ** Does this solution handle outliers and edge cases gracefully?**
- [ ] ** Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -1091,19 +1093,38 @@ public Response apisChangeLifecyclePost(String action, String apiId, String life
...
             } else {
                 WorkflowResponse workflowResponse = RestAPIPublisherUtil.getApiPublisher(username)
                         .updateAPIStatus(apiId, action, lifecycleChecklistMap);
                 response = MappingUtil.toWorkflowResponseDTO(workflowResponse);
+                //if workflow is in pending state or if the executor sends any httpworklfowresponse (workflow state can 
+                //be in either pending or approved state) send back the workflow response 
+                if (WorkflowStatus.CREATED == workflowResponse.getWorkflowStatus()
+                        || workflowResponse instanceof HttpWorkflowResponse) {
+                    URI location = new URI(RestApiConstants.RESOURCE_PATH_APIS + "/" + apiId);
+                    return Response.status(Response.Status.ACCEPTED).header(RestApiConstants.LOCATION_HEADER, location)
```
Comment (Rajith90) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: I think its better to use Response.created(location) to set the location header. Please check if this is re available at msf4j 2.3.0-m2, and update it

Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: Here sending a 202 Accepted response. Response.accepted(location) method is note provided to send a 202 response with location header.

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/applications/add.js [hunk 2]
```diff
@@ -95,8 +95,51 @@ var addApplication = function () {
...
+                                theme : 'relax',
+                                buttons : [
+                                    {addClass: 'btn btn-primary', text: 'Ok', onClick: function ($noty) {
+                                        $noty.close();
+                                        window.location = jsonResponse.redirectUrl;
+                                    }
+                                    },
+                                    {addClass: 'btn btn-danger', text: 'Cancel', onClick: function ($noty) {
+                                        $noty.close();
+                                        window.location = "/store" + data.headers.location;
```
Comment (tgtshanika) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/applications/add.js]: Please use 'contextPath' value to build the redirection URL

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/applications/add.js [hunk 3]
```diff
@@ -95,8 +95,51 @@ var addApplication = function () {
...
+                            noty({
+                                text : "Request has been submitted and is now awaiting approval.",
+                                type : "alert",
+                                dismissQueue: true,
+                                layout : "topCenter",
+                                theme : 'relax',
+                                buttons : [
+                                    {addClass: 'btn btn-primary', text: 'Ok', onClick: function ($noty) {
+                                        $noty.close();
+                                        window.location = "/store" + data.headers.location;
```
Comment (tgtshanika) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/applications/add.js]: Use contextPath instead of '/store'

**Pull request merged**